source/develop/release-management/features/storage/cd-management.md

```diff
@@ -0,0 +1,117 @@
+However, non of these functions worked properly.
+Changing CD failed on block storage as the appropriate volume wasn't activated.
+Ejecting CD failed to deactivate the volume.
+For more details, see following bugs.
+
+### Bugs
+
+- [Error changing CD for a running VM when ISO image is on a block domain](https://bugzilla.redhat.com/1589763)
+- [Volume not deactivated after ejecting CDROM](https://bugzilla.redhat.com/1868643)
+
+
+### Related links
+
+- [Discussion in ovirt devel mailing list.](https://www.mail-archive.com/devel@ovirt.org/msg16082.html)
+- [Vdsm tries to tear down in-use volume of ISO in block storage domain](https://bugzilla.redhat.com/1725915)
+
+
+### Functionality added with this feature
+
+This feature propose new functions to both engine and vdsm for loading, ejecting and changing CD.
+With these functions, all aforementioned functionality will work properly.
+
+#### Vdsm
+
+- Add VM.insertCD API
+- Add VM.ejectCD API
+- Updated recovery flow to check `in_progress` flag for CDs
+
+#### Engine
+
+- Add InsertCD sub-command
+- Add EjectCD sub-command
+- Update ChangeCD to use InsertCD and EjectCD on 4.5 cluster version
+
+
+### Flows
+
+#### VM with empty tray, insert CD
+
+1. engine sends `insertCD` command, specifying ISO by PDIV format
+1. add new drivespec to VM metadata with new attribute `in_progress`
+```
+    <ovirt-vm:device devtype="disk" name="sdc" in_progress="yes">
+```
+1. prepare new drivespec
+  - if failed:
+    1. remove drivespec (log errors)
+    1. fail the flow with original error
+1. loads CD into VM
+  - if failed:
+    1. teardown the volume (log error)
+    1. remove the drive spec (log error)
+    1. fail the flow with the original error
+1. remove `in-progress` attribute from VM metadata
+  - if failed: log the error and succeed
+
+
+#### VM with loaded CD, eject CD
+
+1. engine sends `ejectCD`, specifying ISO path by PDIV format 
+1. vdsm updates VM metadata with new attribute `in_progress`:
+```
+    <ovirt-vm:device devtype="disk" name="sdc" in_progress="yes">
+```
+1. vdsm eject CD from VM (calling libvirt `updateDeviceFlags()`)
+  - if failed (e.g. timeout?):
+    1. remove the `in_progress` flag (log error)
+    1. fail with original error
+1. tear down drivespec
+  - if tearing down volume fails: log error (volume used by another VM)
+1. remove drivespec from VM metadata
+  - if failed: log error and succeed
+
+
+#### VM recovery flow
+
+Try to cleanup storage leftovers from insert and remove operations without changing VM state.
+
+1. metadata in progress, volume is not active
+  - remove metadata from XML
+1. metadata in progress, volume active, no device
+  - deactivate the volume
+  - remove metadata from XML
+1. metadata in progress, volume active, device exists
+  - remove attribute `in_progress` from metadata
+
+
+#### VM with loaded CD, changeCD
+
+1. engine sends ejectCD, specifying ISO path by PDIV format
+1. engine sends insetrCD command, specifying ISO by PDIV format
+
```

Comment:
The internal changes in the system should be here.

Missing info:
- How this feature can be tested?
- Backward compatibility? when new APIs can be used?
- Does it affects upgrades?

Check the feature template for other missing info.